src/core/net/srp_client.cpp

```diff
@@ -885,6 +890,34 @@ Error Client::PrepareUpdateMessage(Message &aMessage)
     return error;
 }
 
+#if OPENTHREAD_CONFIG_PLATFORM_KEY_REFERENCES_ENABLE
+Error Client::ReadOrGenerateKey(Crypto::Ecdsa::P256::KeyPairAsRef &aKeyRef)
+{
+    Error                        error = kErrorNone;
+    Crypto::Ecdsa::P256::KeyPair keyPair;
+
+    VerifyOrExit(!Crypto::Storage::HasKey(aKeyRef.GetKeyRef()));
```

Comment:
The problem with this is that OT factory reset will not remove this key, unless `otPlatSettingsWipe` does it, right? We either need `otPlatSettingsWipe` documented that it should also wipe PSA-stored keys or have an alternative crypto API for wiping the keys. Any thoughts?

Reply:
I fully agree. I think we save other keys and info in PSA for Thread network which should be wiped as well on a factory reset. 

I would suggest adding a new crypto platform API for this and then we can call it from `FactoryReset()` method.

Reply:
Funny you should say that, I am parallely working on an issue where otInstanceFinalize does not clear the PSA keys at all. It is fine in non-PSA use cases, where static buffers are overwritten, but in PSA we will need to clear these manually. I have experimental changes in my local branch, where I have added a new API in key manager and sub mac to clear the keys.

what are your thoughts on this? If it is OK, I can bring those changes to this PR as well.

Reply:
I dont think this can be done in a platform API, because we will need to clear the volatile keys as well, And platform does not store the key refs for these. So either we need to add a new interface in key_manager to clear the keys, or update documentation for `otInstanceFinalize` to istruct this should be followed by a power cycle.

Reply:
I just realized that for SRP not clearing the key on the factory reset is a severe issue because we will not create a new identity. I didn't thought about volatile keys though so if that requires more changes, a separate PR makes sense to me as well. Whatever works for you better :)